sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -160,9 +159,19 @@ trait ExpressionEvalHelper extends ScalaCheckDrivenPropertyChecks with PlanTestB
       expectedErrMsg: String): Unit = {
 
     def checkException(eval: => Unit, testMode: String): Unit = {
+      val modes = if (testMode == "non-codegen mode") {
+        Seq(CodegenObjectFactoryMode.NO_CODEGEN)
+      } else {
+        Seq(CodegenObjectFactoryMode.CODEGEN_ONLY, CodegenObjectFactoryMode.NO_CODEGEN)
```

Comment:
For simplicity, can we always test it with 2 modes?

Reply:
should be OK. since setting mode does not affect `evaluateWithoutCodegen`